src/data/nat/basic.lean

```diff
@@ -1399,4 +1399,62 @@ lemma decreasing_induction_succ_left {P : ℕ → Sort*} (h : ∀n, P (n+1) →
 by { rw [subsingleton.elim mn (le_trans (le_succ m) smn), decreasing_induction_trans,
          decreasing_induction_succ'] }
 
+/-- `log hb n`, is the logarithm of natural number
+`n` in base `b`. If `n` is not a power of `b`, it returns the
+largest `k:ℕ` such that `b^k ≤ n`. -/
+def log (b : ℕ) : ℕ → ℕ
+| n :=
+  if h : b ≤ n ∧ 1 < b then
+    have n / b < n,
+      from div_lt_self
+        (nat.lt_of_lt_of_le (lt_trans zero_lt_one h.2) h.1) h.2,
+    log (n / b) + 1
+  else 0
+
+/-- logarithm in base 2 -/
+def log2 := log 2
+
+/-- logarithm in base 10 -/
+def log10 := log 10
+
```

Comment:
```suggestion
```
Especially with the change to totalize `log`, I don't think these specializations are needed.

Reply:
What if I called them `log₂` and `log₁₀`?